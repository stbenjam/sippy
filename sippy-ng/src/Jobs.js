import { Container, Grid, Paper, Tab, Tabs, Typography } from '@material-ui/core';
import { TabContext } from '@material-ui/lab';
import React from 'react';
import {
    Link, Route, Switch, useRouteMatch
} from "react-router-dom";
import JobsDashboard from './JobsDashboard';
import JobTable from './JobTable';
import PassRateByVariant from './PassRate/passRateByVariant';

export default function Jobs(props) {
    let { path, url } = useRouteMatch();

    return (
        <Route
            path="/"
            render={({ location }) => (
                <TabContext value={path}>
                    <Typography align="center" variant="h4">
                        Job health for {props.release}
                    </Typography>
                    <Grid container justifyContent="center" width="60%" style={{ margin: 20 }}>
                        <Paper>
                            <Tabs
                                value={location.pathname.substring(location.pathname.lastIndexOf('/') + 1)}
                                indicatorColor="primary"
                                textColor="primary"
                            >
                                <Tab label="All jobs" value={props.release} component={Link} to={url} />
                                <Tab label="Jobs by variant" value="variant" component={Link} to={url + "/variant"} />
                                <Tab label="Dashboard" value="dashboard" component={Link} to={url + "/dashboard"} />
                            </Tabs>
                        </Paper>
                    </Grid>
                    <Container size="xl">
                        <Switch>
                            <Route path={path + "/variant"}>
                                <PassRateByVariant release={props.release} />
                            </Route>

                            <Route path={path + "/dashboard"}>
                                <JobsDashboard release={props.release} />
                            </Route>

                            <Route exact path={path}>
                                <JobTable release={props.release} />
                            </Route>
                        </Switch>
                    </Container>
                </TabContext>
            )}
        />
    );
}