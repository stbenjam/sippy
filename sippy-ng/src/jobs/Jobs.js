import PropTypes from 'prop-types'
import { Container, Grid, Paper, Tab, Tabs, Typography } from '@material-ui/core'
import { TabContext } from '@material-ui/lab'
import React, { Fragment } from 'react'

import {
  Link, Route, Switch, useRouteMatch
} from 'react-router-dom'
import SimpleBreadcrumbs from '../components/SimpleBreadcrumbs'
import JobsDetail from './JobsDetail'
import JobTable from './JobTable'
import Variants from './Variants'

/**
 * Jobs is the landing page for jobs with tabs for all jobs, variants,
 * and job runs.
 */
export default function Jobs (props) {
  const { path, url } = useRouteMatch()

  return (
        <Fragment>
            <SimpleBreadcrumbs release={props.release} currentPage="Jobs" />
            <Route
                path="/"
                render={({ location }) => (
                    <TabContext value={path}>
                        <Typography align="center" variant="h4">
                            Job health for {props.release}
                        </Typography>
                        <Grid container justifyContent="center" width="60%" style={{ margin: 20 }}>
                            <Paper>
                                <Tabs
                                    value={location.pathname.substring(location.pathname.lastIndexOf('/') + 1)}
                                    indicatorColor="primary"
                                    textColor="primary"
                                >
                                    <Tab label="All jobs" value={props.release} component={Link} to={url} />
                                    <Tab label="Jobs by variant" value="variant" component={Link} to={url + '/variant'} />
                                    <Tab label="Job runs" value="detail" component={Link} to={url + '/detail'} />
                                </Tabs>
                            </Paper>
                        </Grid>
                        <Container size="xl">
                            <Switch>
                                <Route path={path + '/variant'}>
                                <Variants release={props.release} />
                                </Route>

                                <Route path={path + '/detail'}>
                                    <JobsDetail release={props.release} />
                                </Route>

                                <Route exact path={path}>
                                    <JobTable release={props.release} />
                                </Route>
                            </Switch>
                        </Container>
                    </TabContext>
                )}
            />
        </Fragment>
  )
}

Jobs.propTypes = {
  release: PropTypes.string.isRequired
}
