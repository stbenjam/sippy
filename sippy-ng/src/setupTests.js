/** @jest-environment setup-polly-jest/jest-environment-node */

import '@testing-library/jest-dom'
import Enzyme from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import { Polly } from '@pollyjs/core'
import HttpAdapter from '@pollyjs/adapter-node-http'
import FSPersister from '@pollyjs/persister-fs'
import fetch from 'node-fetch'

// https://github.com/mui-org/material-ui/issues/21293
export const withoutMuiID = (wrapper) => wrapper.html().replace(/id="mui-[0-9]*"/g, '').replace(/aria-labelledby="(mui-[0-9]* *)*"/g, '')

Enzyme.configure({ adapter: new Adapter() })

global.fetch = fetch

// See: https://github.com/vuejs/vue-test-utils/issues/974
global.requestAnimationFrame = cb => cb()
