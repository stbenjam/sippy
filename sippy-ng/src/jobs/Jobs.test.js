/** @jest-environment setup-polly-jest/jest-environment-node */

import 'jsdom-global/register'
import React from 'react'
import { act } from '@testing-library/react'
import { mount } from 'enzyme'
import { setupPolly } from 'setup-polly-jest'
import toJson from 'enzyme-to-json'
import { withoutMuiID } from '../setupTests'
import path from 'path'
import Jobs from './Jobs'
import { BrowserRouter } from 'react-router-dom'

jest.useRealTimers()

describe('Jobs', () => {
  const context = setupPolly({})

  beforeEach(() => {
    context.polly.configure({
      adapters: [require('@pollyjs/adapter-node-http')],
      persister: require('@pollyjs/persister-fs'),
      persisterOptions: {
        fs: {
          recordingsDir: path.resolve(__dirname, '../__recordings__')
        }
      }
    }
    )
  })

  it('should match snapshot', async () => {
    let wrapper
    await act(async () => {
      wrapper = mount(<BrowserRouter><Jobs release="4.9" /></BrowserRouter>)
    })

    expect(wrapper.exists).toBeTruthy()

    // Wait for data to load...
    await act(async () => {
      await new Promise(resolve => setTimeout(resolve, 0))
    })

    wrapper.update()
    expect(withoutMuiID(wrapper)).toMatchSnapshot()
  })
})
