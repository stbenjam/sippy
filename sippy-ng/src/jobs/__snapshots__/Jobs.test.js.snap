// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jobs should match snapshot 1`] = `"<nav class=\\"MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary\\" aria-label=\\"breadcrumb\\"><ol class=\\"MuiBreadcrumbs-ol\\"><li class=\\"MuiBreadcrumbs-li\\"><a href=\\"/release/4.9\\">Overview</a></li><li aria-hidden=\\"true\\" class=\\"MuiBreadcrumbs-separator\\">/</li><li class=\\"MuiBreadcrumbs-li\\"><p class=\\"MuiTypography-root MuiTypography-body1\\">Jobs</p></li></ol></nav>"`;
